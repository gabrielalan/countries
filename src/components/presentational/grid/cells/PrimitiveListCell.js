import React from 'react';
import TextCell from './TextCell';

/**
 * A Cell type that can iterate over a list of primitive values (string, numbers, etc)
 * And render a certain type with it.
 */
export default function PrimitiveListCell({ CellType, children }) {
  return children.map(data => <CellType key={data}>{data}</CellType>);
}

PrimitiveListCell.defaultProps = {
  CellType: TextCell
};

/**
 * HighOrder function that returns a component
 * Patching the PrimitiveListCell to contain a CellType
 */
export const withCellType = type => props => <PrimitiveListCell CellType={type} {...props} />;