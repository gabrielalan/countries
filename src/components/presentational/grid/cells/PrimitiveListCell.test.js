import React from 'react';
import renderer from 'react-test-renderer';
import PrimitiveListCell from './PrimitiveListCell';

describe('PrimitiveListCell', () => {
  it('should render text normally', () => {
    const tree = renderer.create(<PrimitiveListCell>{['a', 'b', 'c']}</PrimitiveListCell>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render text normally', () => {
    const tree = renderer.create(<PrimitiveListCell>{[1, 2, 3]}</PrimitiveListCell>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
