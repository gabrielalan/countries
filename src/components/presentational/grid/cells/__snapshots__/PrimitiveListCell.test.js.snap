// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimitiveListCell should render text normally 1`] = `
Array [
  "a",
  "b",
  "c",
]
`;

exports[`PrimitiveListCell should render text normally 2`] = `
Array [
  "1",
  "2",
  "3",
]
`;
